File: 민성/2631.py
# BaekJoon 05/10 2023
# 2631 줄세우기
# 13:31 ~ 14:00

N = int(input())
lines = [int(input()) for _ in range(N)]

dp = [1] * N

for i in range(1, N):
    for j in range(0, i):
        if lines[j] < lines[i]:
            dp[i] = max(dp[i], dp[j]+1)

print(N - max(dp))

File: 동규/2631.js
const fs = require("fs");
const path = process.platform === "linux" ? "/dev/stdin" : "example.txt";
const input = fs.readFileSync(path).toString().trim().split("\n");

const N = +input[0];
const idle = input.slice(1, N + 1).map(Number);
const dp = Array(N + 1).fill(1); // 1 이상

function dpLis() {
  let max = Number.MIN_SAFE_INTEGER;

  for (let i = 1; i < N; i++) {
    for (let j = 0; j < i; j++) {
      if (idle[j] < idle[i]) {
        dp[i] = Math.max(dp[i], dp[j] + 1);
      }
    }
    max = Math.max(max, dp[i]);
  }

  return max;
}

console.log(N - dpLis());
